parser_library/src/parsing/grammar/label_field_rules.g4

```diff
@@ -15,97 +15,127 @@
-	{
-		collector.add_hl_symbol(token_info(provider.get_range( $l_char_string_sp.ctx),hl_scopes::label));
-		collector.add_hl_symbol(token_info(provider.get_range( $l_char_string_o.ctx),hl_scopes::label));
-		$l_char_string_sp.value.append(std::move($l_char_string_o.value));
-		auto r = provider.get_range( $l_char_string_sp.ctx->getStart(),$l_char_string_o.ctx->getStop());
-		collector.set_label_field(std::move($l_char_string_sp.value),r);
-	}
-	| l_model
-	{
-		collector.set_label_field(std::move($l_model.chain),provider.get_range( $l_model.ctx));
-	}	//model stmt rule with no space
-	| l_model_sp
-	{
-		collector.set_label_field(std::move($l_model_sp.chain),provider.get_range( $l_model_sp.ctx));
-	}	//model stmt rule with possible space
-	|
-	{
-		collector.set_label_field(provider.get_empty_range( _localctx->getStart()));
-	};
-
-
-l_common_rules [concat_chain* chain]
-	: l_string_v_apo_sp[$chain] l_string_v[$chain]
-	| l_string_poss_space_c
-	{
-		$chain->emplace_back(char_str_conc(std::move($l_string_poss_space_c.value), provider.get_range($l_string_poss_space_c.ctx)));
-	}
-	l_string_v[$chain]
-	| l_string_v_apo_sp[$chain] l_string
-	{
-		$chain->emplace_back(char_str_conc(std::move($l_string.value), provider.get_range($l_string.ctx)));
-	};
-
-l_model_sp returns [concat_chain chain]
-	: l_string_v_apo_sp[&$chain]
-	| l_common_rules[&$chain]
-	| l_common_rules[&$chain] l_string_v_apo[&$chain]
-	| l_common_rules[&$chain] l_string_no_space_c
-	{
-		$chain.emplace_back(char_str_conc(std::move($l_string_no_space_c.value), provider.get_range($l_string_no_space_c.ctx)));
-	}
-	| l_string_poss_space_c l_string
-	{
-		$chain.emplace_back(char_str_conc(std::move($l_string_poss_space_c.value), provider.get_range($l_string_poss_space_c.ctx)));
-		$chain.emplace_back(char_str_conc(std::move($l_string.value), provider.get_range($l_string.ctx)));
-	}
-	l_string_v_apo[&$chain]
 	;
-	finally
-	{concatenation_point::clear_concat_chain($chain);}
-
-l_model returns [concat_chain chain]
-	: l_string_v[&$chain]
-	| l_string
-	{
-		$chain.emplace_back(char_str_conc(std::move($l_string.value), provider.get_range($l_string.ctx)));
-	}
-	l_string_v_apo[&$chain]
-	| l_string_v[&$chain] l_string_v_apo[&$chain]
-	| l_string_v[&$chain] l_string_no_space_c
-	{
-		$chain.emplace_back(char_str_conc(std::move($l_string_no_space_c.value), provider.get_range($l_string_no_space_c.ctx)));
-	};
-	finally
-	{concatenation_point::clear_concat_chain($chain);}
-
 
+label_string [concat_chain* chain, std::string* buffer, bool* has_variables, bool allow_space]
+	:
+	ap1=(APOSTROPHE | ATTR) {add_label_component($ap1, *$chain, *$buffer, *$has_variables);}
+	(
+		{!$allow_space && _input->LA(1)==SPACE || is_attribute_consuming(_input->LT(-2))}?
+		|
+		(
+			t=(
+				SPACE|
+				DOT|
+				ASTERISK|
+				MINUS|
+				PLUS|
+				LT|
+				GT|
+				COMMA|
+				LPAR|
+				RPAR|
+				SLASH|
+				EQUALS|
+				ORDSYMBOL|
+				IDENTIFIER|
+				OR|
+				AND|
+				EQ|
+				LE|
+				LTx|
+				GTx|
```

Comment:
I guess these are needed but I actually don't know what they are. Should it be just `LT` and `GT` (without `x`)?

Reply:
As discussed, they are not needed, I will do the cleanup PR as the next step and remove all these unused token IDs.